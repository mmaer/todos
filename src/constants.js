export const LOCAL_STORAGE_KEY = "todos";

export const ENTER_KEY = 13;
export const ESCAPE_KEY = 27;

export const FILTER_ALL = "all";
export const FILTER_ACTIVE = "active";
export const FILTER_COMPLETED = "completed";

export const FILTERS = [FILTER_ALL, FILTER_ACTIVE, FILTER_COMPLETED];
