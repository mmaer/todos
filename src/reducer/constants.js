export const ADD_TODO = "ADD_TODO";
export const REMOVE_TODO = "REMOVE_TODO";
export const EDIT_TODO_TITLE = "EDIT_TODO_TITLE";

export const TOGGLE_COMPLETED_STATUS = "TOGGLE_COMPLETED_STATUS";
export const TOGGLE_EDITING_STATUS = "TOGGLE_EDITING_STATUS";

export const REMOVE_COMPLETED_TODOS = "REMOVE_COMPLETED_TODOS";
