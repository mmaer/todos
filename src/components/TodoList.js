import React from "react";
import Todo from "./Todo";

import { FILTER_ACTIVE, FILTER_COMPLETED, FILTER_ALL } from "../constants";

const FILTERS = {
  [FILTER_ACTIVE]: completed => !completed,
  [FILTER_COMPLETED]: completed => completed,
  [FILTER_ALL]: () => true
};

const TodoList = ({ filter, ...props }) => (
  <ul className="todo-list">
    {props.todos
      .filter(({ completed }) => FILTERS[filter](completed))
      .map((todo, id) => (
        <Todo {...props} todo={todo} id={id} key={id} />
      ))}
  </ul>
);

export default TodoList;
