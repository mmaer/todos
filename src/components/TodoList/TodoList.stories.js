import React from "react";

import { FILTER_ALL } from "../../constants";

import TodoList from ".";

export default {
  title: "TodoList"
};

const TODOS = [
  { title: "Todo1", editing: false, completed: false, id: 1 },
  { title: "Todo2", editing: false, completed: false, id: 1 }
];

export const todoList = () => <TodoList todos={TODOS} filter={FILTER_ALL} />;
