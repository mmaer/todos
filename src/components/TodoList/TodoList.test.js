import React from "react";
import { mount } from "enzyme";

import { FILTER_ACTIVE, FILTER_COMPLETED, FILTER_ALL } from "../../constants";

import TodoList from "./";

const TODOS = [
  { completed: false },
  { completed: false },
  { completed: true },
  { completed: false },
  { completed: true },
  { completed: false },
  { completed: false },
];

describe("TodoList component", () => {
  it("Should match snapshot", () => {
    const wrapper = mount(<TodoList todos={TODOS} filter={FILTER_ALL} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("Should display all todos", () => {
    const wrapper = mount(<TodoList todos={TODOS} filter={FILTER_ALL} />);
    const todos = wrapper.find("li");
    expect(todos.length).toBe(7);
  });

  it("Should display only active todos", () => {
    const wrapper = mount(<TodoList todos={TODOS} filter={FILTER_ACTIVE} />);
    const todos = wrapper.find("li");
    expect(todos.length).toBe(5);
  });

  it("Should display only completed todos", () => {
    const wrapper = mount(<TodoList todos={TODOS} filter={FILTER_COMPLETED} />);
    const todos = wrapper.find("li");
    expect(todos.length).toBe(2);
  });
});
