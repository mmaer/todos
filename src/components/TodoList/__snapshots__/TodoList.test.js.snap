// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList component Should match snapshot 1`] = `
<TodoList
  filter="all"
  todos={
    Array [
      Object {
        "completed": false,
      },
      Object {
        "completed": false,
      },
      Object {
        "completed": true,
      },
      Object {
        "completed": false,
      },
      Object {
        "completed": true,
      },
      Object {
        "completed": false,
      },
      Object {
        "completed": false,
      },
    ]
  }
>
  <ul
    className="todo-list"
  >
    <Memo()
      completed={false}
      id={0}
      key="0"
    >
      <li
        className="todo"
      >
        <Memo()
          completed={false}
          id={0}
        >
          <div
            className="todo-view"
            onDoubleClick={[Function]}
          >
            <input
              checked={false}
              className="todo-view__toggle"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="todo-view__label"
            />
            <button
              className="todo-view__remove"
              onClick={[Function]}
            />
          </div>
        </Memo()>
      </li>
    </Memo()>
    <Memo()
      completed={false}
      id={1}
      key="1"
    >
      <li
        className="todo"
      >
        <Memo()
          completed={false}
          id={1}
        >
          <div
            className="todo-view"
            onDoubleClick={[Function]}
          >
            <input
              checked={false}
              className="todo-view__toggle"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="todo-view__label"
            />
            <button
              className="todo-view__remove"
              onClick={[Function]}
            />
          </div>
        </Memo()>
      </li>
    </Memo()>
    <Memo()
      completed={true}
      id={2}
      key="2"
    >
      <li
        className="todo"
      >
        <Memo()
          completed={true}
          id={2}
        >
          <div
            className="todo-view"
            onDoubleClick={[Function]}
          >
            <input
              checked={true}
              className="todo-view__toggle"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="todo-view__label todo-view__label--completed"
            />
            <button
              className="todo-view__remove"
              onClick={[Function]}
            />
          </div>
        </Memo()>
      </li>
    </Memo()>
    <Memo()
      completed={false}
      id={3}
      key="3"
    >
      <li
        className="todo"
      >
        <Memo()
          completed={false}
          id={3}
        >
          <div
            className="todo-view"
            onDoubleClick={[Function]}
          >
            <input
              checked={false}
              className="todo-view__toggle"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="todo-view__label"
            />
            <button
              className="todo-view__remove"
              onClick={[Function]}
            />
          </div>
        </Memo()>
      </li>
    </Memo()>
    <Memo()
      completed={true}
      id={4}
      key="4"
    >
      <li
        className="todo"
      >
        <Memo()
          completed={true}
          id={4}
        >
          <div
            className="todo-view"
            onDoubleClick={[Function]}
          >
            <input
              checked={true}
              className="todo-view__toggle"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="todo-view__label todo-view__label--completed"
            />
            <button
              className="todo-view__remove"
              onClick={[Function]}
            />
          </div>
        </Memo()>
      </li>
    </Memo()>
    <Memo()
      completed={false}
      id={5}
      key="5"
    >
      <li
        className="todo"
      >
        <Memo()
          completed={false}
          id={5}
        >
          <div
            className="todo-view"
            onDoubleClick={[Function]}
          >
            <input
              checked={false}
              className="todo-view__toggle"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="todo-view__label"
            />
            <button
              className="todo-view__remove"
              onClick={[Function]}
            />
          </div>
        </Memo()>
      </li>
    </Memo()>
    <Memo()
      completed={false}
      id={6}
      key="6"
    >
      <li
        className="todo"
      >
        <Memo()
          completed={false}
          id={6}
        >
          <div
            className="todo-view"
            onDoubleClick={[Function]}
          >
            <input
              checked={false}
              className="todo-view__toggle"
              onChange={[Function]}
              type="checkbox"
            />
            <label
              className="todo-view__label"
            />
            <button
              className="todo-view__remove"
              onClick={[Function]}
            />
          </div>
        </Memo()>
      </li>
    </Memo()>
  </ul>
</TodoList>
`;
